### PR TITLE
Add support for hook_menu() load arguments

### PR DESCRIPTION
**Pull request by darrenoh**

### CODE REVIEW

**Comment by darrenoh** [comment]
Found crashes on access checks because request object is not available. Reworking.

File: src/Access/CustomControllerAccessCallback.php [hunk 1]
```diff
@@ -29,6 +29,11 @@ public function check(RouteMatchInterface $route_match): AccessResultInterface
                 $route_match->getRouteName()
             ));
         }
+        foreach ($access_arguments as &$argument) {
+            if (is_string($argument) && $placeholder = preg_filter('/(^{)(.*)(}$)/', '$2', $argument)) {
```
Comment (mglaman) [inline src/Access/CustomControllerAccessCallback.php:33]: Can you help explain this regex? I'm trying to parse it, but `_custom_access_arguments` is from `access arguments`. It should just be an `array_map` to get the values from the route match.

Edit: I read this before `src/Routing/HookMenuRoutes.php` changes.

**Review by mglaman — COMMENTED** [review]
Reading this through, it makes sense. I'm curious on the preg_filter

It also looks like the include path wasn't properly working since `file path` could be provided.

Looks like we should maybe split that out, because Drupal 7 calculated the vale of `file` and `file path` to `include file` with the full path

```php
    // Calculate out the file to be included for each callback, if any.
    if ($item['file']) {
      $file_path = $item['file path'] ? $item['file path'] : drupal_get_path('module', $item['module']);
      $item['include file'] = $file_path . '/' . $item['file'];
    }
```

File: src/Access/CustomControllerAccessCallback.php (hunk 1 above)
Comment (darrenoh) [reply src/Access/CustomControllerAccessCallback.php:33]: Right, I enclosed access arguments in curly braces if they need to be processed with a load function.

**Review by mglaman — APPROVED** [review]

**Pull request merged**